src/Aspire.Dashboard/Components/Dialogs/SettingsDialog.razor.cs

```diff
@@ -44,6 +70,21 @@ private async Task SettingChangedAsync()
         }
     }
 
+    private void OnLanguageChanged()
+    {
+        if (_selectedUiCulture is null || StringComparers.Culture.Equals(CultureInfo.CurrentUICulture.Name, _selectedUiCulture.Name))
+        {
+            return;
+        }
+
+        var uri = new Uri(NavigationManager.Uri)
+            .GetComponents(UriComponents.PathAndQuery, UriFormat.Unescaped);
+
+        NavigationManager.NavigateTo(
+            DashboardUrls.SetLanguageUrl(_selectedUiCulture.Name, uri),
+            forceLoad: true);
```

Comment:
Add a comment here that a cookie must be set and the dashboard reloaded to use the new culture set by localization middle.